lang/java/reef-runtime-spark/src/test/java/org/apache/reef/runtime/spark/job/ReefOnSpark.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runtime.spark.job;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.reef.client.DriverConfiguration;
+import org.apache.reef.tang.exceptions.InjectionException;
+import org.apache.reef.proto.ReefServiceProtos;
+import org.apache.reef.client.DriverLauncher;
+import org.apache.reef.runtime.common.REEFEnvironment;
+import org.apache.reef.runtime.yarn.client.unmanaged.UnmanagedAmYarnClientConfiguration;
+import org.apache.reef.runtime.yarn.client.unmanaged.UnmanagedAmYarnDriverConfiguration;
+import org.apache.reef.util.EnvironmentUtils;
+import org.apache.spark.SparkConf;
+import org.apache.reef.tang.Configuration;
+import org.apache.spark.SparkContext;
+
+// Run:
+// ..\spark\bin\spark-submit.cmd
+//     --master yarn --deploy-mode cluster
+//     --class org.apache.reef.examples.hellospark.ReefOnSpark
+//     .\target\reef-examples-spark-0.16.0-SNAPSHOT-shaded.jar
+
+
+/**
+ * Abstraction around submitting a reef job on spark.
+ */
+public final class ReefOnSpark {
+
+  private Logger logger = Logger.getLogger(ReefOnSpark.class.getName());
+
+  private static final String ROOTFOLDER = ".";
+
+  private static final Configuration RUNTIME_CONFIG = UnmanagedAmYarnClientConfiguration.CONF
```

Comment:
That stuff we won't need - I assume that in the final version you'll have something like `SparkClientConfiguration` that assumes that the Spark job is already running. That is, a REEF Spark runtime will just launch a Driver thread to communicate with REEF Evaluators (running, in turn, on Spark Executors) and handle the corresponding events both from REEF and from the outer Spark app.